### PR TITLE
Change GraphQl call to a single resource to call ES rather than AR

### PR DESCRIPTION
**Pull request by kjgarza**
## Purpose

Improve performance on single DOI requests in GRaphQL

closes:  https://github.com/datacite/datacite/issues/1329

## Approach

Call ES rather than AR. AR makes additional calls for relationships as mentioned in https://github.com/datacite/datacite/issues/1329 that make everything slow

#### Open Questions and Pre-Merge TODOs

- How large is the performance increase? 
- A: Potentially around 5 seconds  (order of Sec.) in some cases for DOIs with references/citations/parts/etc
- Would performance DOIs request without any relationship will be affected?
- A: That would potentially increase in the other of mSec.  Benchmarking would be interesting to do.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def execute
         query,
         variables: variables, context: context, operation_name: operation_name,
       )
-    render json: ApolloFederation::Tracing.attach_trace_to_result(result)
```
[the remark this answers is not in the document]
Comment (kjgarza) [reply app/controllers/graphql_controller.rb:19]: according to this , we do not https://github.com/Gusto/apollo-federation-ruby/blob/1d3baf4f8efcd02e7bf5bc7e3fee5b4fb963cd25/lib/apollo-federation/tracing.rb#L18

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -1217,9 +1217,11 @@ def set_doi(id)
     doi = doi_from_url(id)
     fail ActiveRecord::RecordNotFound if doi.nil?
 
-    doi = DataciteDoi.where(doi: doi, aasm_state: "findable").first
-    fail ActiveRecord::RecordNotFound if doi.nil?
-
-    doi
+    # doi = DataciteDoi.where(doi: doi, aasm_state: "findable").first
```
[the remark this answers is not in the document]
Comment (kjgarza) [reply app/graphql/types/query_type.rb:1220]: Well, if things go tits-up we can return it quickly. but, sure I can clean it

[the remark this answers is not in the document]
Comment (richardhallett) [reply app/graphql/types/query_type.rb:1220]: That's what git is for, git revert #commithash

File: app/models/concerns/indexable.rb [hunk 3]
```diff
@@ -107,6 +107,7 @@ def send_contact_export_message(data)
     # shoryuken_class is needed for the consumer to process the message
     # we use the AWS SQS client directly as there is no consumer in this app
     def send_message(body, options = {})
+      return nil if Rails.env.development?
```
Comment (richardhallett) [inline app/models/concerns/indexable.rb:110]: Why? We still need this to function on development right?

File: app/graphql/types/query_type.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjgarza) [reply app/graphql/types/query_type.rb]: done

File: spec/graphql/types/work_type_spec.rb [hunk 4]
```diff
@@ -307,7 +307,7 @@
         [
           {
             "locations" => [{ "column" => 9, "line" => 2 }],
-            "message" => "Record not found",
+            "message" => "undefined method `[]' for nil:NilClass",
```
[the remark this answers is not in the document]
Comment (kjgarza) [reply spec/graphql/types/work_type_spec.rb]: done

**Review by richardhallett — APPROVED** [review]

File: app/models/concerns/indexable.rb (hunk 3 above)
Comment (kjgarza) [reply app/models/concerns/indexable.rb:110]: we will revisit this in a different PR and issues

**Pull request closed without merging**